interfaces/datastructure/GlobalDescriptor.h

```diff
@@ -0,0 +1,128 @@
+/**
+ * @copyright Copyright (c) 2024 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @file */
+
+#ifndef SOLAR_GLOBALDESCRIPTOR_H
+#define SOLAR_GLOBALDESCRIPTOR_H
+
+#include <core/Messages.h>
+#include <core/SerializationDefinitions.h>
+#include <core/SolARFrameworkDefinitions.h>
+#include <datastructure/DescriptorBuffer.h>
+#include <optional>
+
+namespace SolAR {
+namespace datastructure {
+
+/**
+ * @enum global image descriptor type
+*/
+enum class GlobalDescriptorType {
+    NETVLAD, /**< NetVLAD image descriptor */
+};
+
+/**
+ * @enum data type of the global image descriptor 
+*/
+enum class GlobalDescriptorDataType : size_t {
+    TYPE_8U = 1,  /**< each global descriptor is stored in one byte. */
+    TYPE_32F = 4, /**< each global descriptor is stored in four bytes. */
+};
+
+/**
+ * @class global image descriptor
+*/
+class SOLARFRAMEWORK_API GlobalDescriptor {
+public:
+    /**
+     * @brief builder of global descriptor 
+     * @param[in] type global descriptor type 
+     * @param[in] buffer data pointer 
+     * @return shared pointer to GlobalDescriptor if success, otherwise std::nullopt 
+    */
+    static std::optional<GlobalDescriptor> build(GlobalDescriptorType type, unsigned char* buffer);
+
+    /**
+     * @brief length of the descriptor, i.e. number of elements stored in the global descriptor
+     * @return number of elements in the global descriptor 
+    */
+    size_t getLength() const;
+
+    /**
+     * @brief const pointer to the descriptor buffer
+     * @return const pointer to the descriptor buffer 
+    */
+    const unsigned char* data() const;
+
+    /**
+     * @brief non-const pointer to the descriptor buffer
+     * @return non-const pointer to data buffer 
+    */
+    unsigned char* data();
+
+    /**
+     * @brief get descriptor type 
+     * @return the type of global descriptor  
+    */
+    GlobalDescriptorType getType() const;
+    
+    /**
+     * @brief get descriptor data type 
+     * @return the data type of the global descriptor 
+    */
+    GlobalDescriptorDataType getDataType() const;
+
+    /**
+     * @brief check if content is valid
+     * @return boolean
+    */
+    bool isValid(); 
+
+private:
+    /**
+     * @brief default constructor 
+    */
+    GlobalDescriptor() = default;
```

Comment:
Remove (is not used, right?)

Reply:
After removing it, solarframework does not compile any more (). "No matching function for call to solar::datastructure::GlobalDescriptor::GlobalDescriptor()  (Frame.cpp:226 IMPLEMENTSERIALIZE(Frame))"  it seems that the default constructor is used for boost serialization,

Reply:
Ah ok then.